lib/web/FetchCompileWasmMainTemplatePlugin.js

```diff
@@ -70,27 +157,26 @@ class FetchCompileWasmMainTemplatePlugin {
 						"var installedWasmModuleData = installedWasmModules[wasmModuleId];",
 						"",
 						'// a Promise means "currently loading" or "already loaded".',
-						"promises.push(installedWasmModuleData ||",
 						Template.indent([
-							`(installedWasmModules[wasmModuleId] = fetch(${
-								mainTemplate.requireFn
-							}.p + ${wasmModuleSrcPath}).then(function(response) {`,
+							`var importObject = importObjects[wasmModuleId]`,
+							`var req = fetch(${mainTemplate.requireFn}.p + ${
+								wasmModuleSrcPath
+							})`,
+							"if(WebAssembly.instantiateStreaming) {",
+							Template.indent([
+								"promises.push(WebAssembly.instantiateStreaming(req, importObject)",
+								`.then(function(res) { ${
+									mainTemplate.requireFn
+								}.w[wasmModuleId] = installedWasmModules[wasmModuleId] = res.instance;`,
+								"}))"
+							]),
+							"} else {",
 							Template.indent([
-								"if(WebAssembly.compileStreaming) {",
-								Template.indent([
-									"return WebAssembly.compileStreaming(response);"
-								]),
-								"} else {",
-								Template.indent([
-									"return response.arrayBuffer().then(function(bytes) { return WebAssembly.compile(bytes); });"
-								]),
-								"}"
+								// FIXME(sven): ensrue this still works / change it
+								"promises.push(response.arrayBuffer().then(function(bytes) { installedWasmModules[wasmModuleId] = WebAssembly.compile(bytes); }));"
```

Comment:
`WebAssembly.compile` -> `WebAssembly.instantiate`

It returns a promise and the result need to be stored in installedWasmModules

Reply:
Yeah I didn't change this branch, that's why I added the FIXME. I'll fix this at the end so I can also test it